data/data_loader.py

```diff
@@ -31,68 +20,37 @@ def create_dataloaders(
     CustomerIdMap,
     ArticleIdMap,
 ]:
-    data = torch.load("data/derived/graph.pt")
-    # Add a reverse ('article', 'rev_buys', 'customer') relation for message passing:
-    data = T.ToUndirected()(data)
-
-    # Perform a link-level split into training, validation, and test edges:
-    train_split, val_split, test_split = T.RandomLinkSplit(
-        num_val=config.val_split,
-        num_test=config.test_split,
-        neg_sampling_ratio=0.5,
-        add_negative_train_samples=True,
-        edge_types=[("customer", "buys", "article")],
-        rev_edge_types=[("article", "rev_buys", "customer")],
-        is_undirected=True,
-    )(data)
-    # when neg_sampling_ratio > 0 and add_negative_train_samples=True only then you will have negative edges
-
-    train_loader = LinkNeighborLoader(
-        train_split,
-        num_neighbors=[config.num_neighbors] * config.num_neighbors_it,
-        batch_size=config.batch_size,
-        edge_label_index=(
-            ("customer", "buys", "article"),
-            train_split[("customer", "buys", "article")].edge_label_index,
-        ),
-        edge_label=train_split[("customer", "buys", "article")].edge_label,
-        directed=False,
-        replace=False,
-        shuffle=True,
-        num_workers=config.num_workers,
-        pin_memory=True,
+    data_dir = "data/derived/"
+    train_dataset = GraphDataset(
+        edge_dir=data_dir + "edges_train.pt", graph_dir=data_dir + "train_graph.pt"
     )
-    val_loader = LinkNeighborLoader(
-        val_split,
-        num_neighbors=[64] * 2,
-        batch_size=config.batch_size,
-        edge_label_index=(
-            ("customer", "buys", "article"),
-            val_split[("customer", "buys", "article")].edge_label_index,
-        ),
-        edge_label=val_split[("customer", "buys", "article")].edge_label,
-        directed=False,
-        replace=False,
-        shuffle=True,
-        num_workers=config.num_workers,
-        pin_memory=True,
+    val_dataset = GraphDataset(
+        edge_dir=data_dir + "edges_val.pt", graph_dir=data_dir + "val_graph.pt"
     )
-    test_loader = LinkNeighborLoader(
-        test_split,
-        num_neighbors=[64] * 2,
-        batch_size=config.batch_size,
-        edge_label_index=(
-            ("customer", "buys", "article"),
-            test_split[("customer", "buys", "article")].edge_label_index,
-        ),
-        edge_label=test_split[("customer", "buys", "article")].edge_label,
-        directed=False,
-        replace=False,
-        shuffle=True,
-        num_workers=config.num_workers,
-        pin_memory=True,
+    test_dataset = GraphDataset(
+        edge_dir=data_dir + "edges_test.pt", graph_dir=data_dir + "test_graph.pt"
     )
 
+    # Add a reverse ('article', 'rev_buys', 'customer') relation for message passing:
```

Comment:
will we need this?

Reply:
We do this manually, so I think we can take it out.